vald/evm/evm.go

```diff
@@ -246,77 +250,81 @@ func (mgr Mgr) ProcessTransferKeyConfirmation(event *types.ConfirmKeyTransferSta
-	_, err = mgr.broadcaster.Broadcast(context.TODO(), voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain, events...)))
-
-	return err
+	return events
 }
 
 // ProcessGatewayTxConfirmation votes on the correctness of an EVM chain gateway's transactions
 func (mgr Mgr) ProcessGatewayTxConfirmation(event *types.ConfirmGatewayTxStarted) error {
-	if !mgr.isParticipantOf(event.Participants) {
-		mgr.logger("pollID", event.PollID).Debug("ignoring gateway tx confirmation poll: not a participant")
-		return nil
-	}
-
-	txReceipt, err := mgr.GetTxReceiptIfFinalized(event.Chain, common.Hash(event.TxID), event.ConfirmationHeight)
-	if err != nil {
-		return err
-	}
-	if txReceipt == nil {
-		mgr.logger().Infof("broadcasting empty vote for poll %s", event.PollID.String())
-		_, err := mgr.broadcaster.Broadcast(context.TODO(), voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain)))
-
-		return err
-	}
-
-	events := mgr.processGatewayTxLogs(event.Chain, event.GatewayAddress, txReceipt.Logs)
-	mgr.logger().Infof("broadcasting vote %v for poll %s", events, event.PollID.String())
-	_, err = mgr.broadcaster.Broadcast(context.TODO(), voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain, events...)))
-
-	return err
+	mappedEvent := &types.ConfirmGatewayTxsStarted{
+		PollMappings:       []types.PollMapping{{PollID: event.PollID, TxID: event.TxID}},
+		Chain:              event.Chain,
+		GatewayAddress:     event.GatewayAddress,
+		ConfirmationHeight: event.ConfirmationHeight,
+		Participants:       event.Participants,
+	}
+	return mgr.ProcessGatewayTxsConfirmation(mappedEvent)
 }
 
 // ProcessGatewayTxsConfirmation votes on the correctness of an EVM chain multiple gateway transactions
 func (mgr Mgr) ProcessGatewayTxsConfirmation(event *types.ConfirmGatewayTxsStarted) error {
-	if !mgr.isParticipantOf(event.Participants) {
-		pollIDs := slices.Map(event.PollMappings, func(m types.PollMapping) vote.PollID { return m.PollID })
-		mgr.logger("poll_ids", pollIDs).Debug("ignoring gateway txs confirmation poll: not a participant")
+	vc := voteContext{
+		Participants:       event.Participants,
+		PollMappings:       event.PollMappings,
+		Chain:              event.Chain,
+		ConfirmationHeight: event.ConfirmationHeight,
+		PollType:           "gateway txs confirmation",
+	}
+	return mgr.vote(vc, func(logs []*geth.Log) []types.Event {
+		return mgr.processGatewayTxLogs(event.Chain, event.GatewayAddress, logs)
+	})
+}
+
+type voteContext struct {
+	PollType           string
+	Participants       []sdk.ValAddress
+	PollMappings       []types.PollMapping
+	Chain              nexus.ChainName
+	ConfirmationHeight uint64
+}
+
+func (mgr Mgr) vote(voteContext voteContext, logsToVotes func(logs []*geth.Log) []types.Event) error {
+	if !mgr.isParticipantOf(voteContext.Participants) {
+		pollIDs := slices.Map(voteContext.PollMappings, func(m types.PollMapping) vote.PollID { return m.PollID })
```

Comment:
```suggestion
		pollIDs := slices.Map(voteContext.PollMappings, types.PollMapping.GetPollID)
```
add a getter perhaps?